### PR TITLE
Metric

### PR DESCRIPTION
**Pull request by sun765**

### CODE REVIEW

File: packages/render-fidelity-tools/src/workflows/test-fidelity.ts [hunk 1]
```diff
@@ -22,17 +22,19 @@ const require = module.createRequire(import.meta.url);
 
 import {ArtifactCreator} from '../artifact-creator.js';
 
+
 const configPath = resolve(process.argv[2]);
 const rootDirectory = resolve(dirname(configPath));
 const config = require(configPath);
 
 const outputDirectory = join(rootDirectory, 'results');
+const portal = 9030;
```
Comment (elalish) [inline packages/render-fidelity-tools/src/workflows/test-fidelity.ts:31]: Were these changes intended to be committed?

File: packages/render-fidelity-tools/test/config.json [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{
+{
```
Comment (elalish) [inline packages/render-fidelity-tools/test/config.json:1]: Small white-space changes like this usually indicate that clang-format hasn't been applied. If you resave this with clang-format enabled, I think this change will disappear.

File: packages/render-fidelity-tools/src/artifact-creator.ts [hunk 3]
```diff
@@ -165,7 +166,7 @@ export class ArtifactCreator {
   async captureScreenshot(
       renderer: string, scenarioName: string, dimensions: Dimensions,
       outputPath: string = join(this.outputDirectory, 'model-viewer.png')) {
-    const devicePixelRatio = 2;
+    const devicePixelRatio = 2;  // hard code?
```
Comment (elalish) [inline packages/render-fidelity-tools/src/artifact-creator.ts:169]: This is the DPR of our macbooks. It doesn't really make any difference, except it exercises our DPR codepath, so helps to catch more bugs.

File: packages/render-fidelity-tools/src/artifact-creator.ts [hunk 4]
```diff
@@ -56,7 +56,7 @@ export class ArtifactCreator {
 
     for (const scenarioBase of scenarios) {
       const scenarioName = scenarioBase.name;
-      const scenario = this[$configReader].scenarioConfig(scenarioName)!;
+      const scenario = this[$configReader].scenarioConfig(scenarioName)!;  // ??
```
Comment (elalish) [inline packages/render-fidelity-tools/src/artifact-creator.ts:59]: Do you have a question for me?

File: packages/render-fidelity-tools/src/artifact-creator.ts [hunk 5]
```diff
@@ -70,7 +70,8 @@ export class ArtifactCreator {
       mkdirp.sync(scenarioOutputDirectory);
 
       const screenshot = await this.captureScreenshot(
-          'model-viewer',
+          'model-viewer',  // hard code to model viewer?  how do you load other
+                           // renders?
```
Comment (elalish) [inline packages/render-fidelity-tools/src/artifact-creator.ts:74]: This is for the captureAndAnalyze function, which we currently only run on model-viewer (that's the only one we generate statistics for). As we add other renderers, and especially when we have better ground truth renders to compare to, we should revisit how these scripts should operate.

File: packages/render-fidelity-tools/src/workflows/test-fidelity.ts (hunk 1 above)
Comment (sun765) [reply packages/render-fidelity-tools/src/workflows/test-fidelity.ts:31]: Yeah, do you think it's a good way to refactor code?

File: packages/render-fidelity-tools/src/artifact-creator.ts (hunk 4 above)
Comment (sun765) [reply packages/render-fidelity-tools/src/artifact-creator.ts:59]: just ignore it, i didn't mean to commit it. i don't understand the syntax at first so I leave a mark here. I have already learned what does this mean. I will remove this in the new push

File: packages/render-fidelity-tools/src/workflows/test-fidelity.ts (hunk 1 above)
Comment (elalish) [reply packages/render-fidelity-tools/src/workflows/test-fidelity.ts:31]: maybe just call it port; that's more commonly used.

File: packages/render-fidelity-tools/src/workflows/test-fidelity.ts [hunk 6]
```diff
@@ -43,6 +44,7 @@ try {
 
 let scenarioWhitelist: Set<string>|null = null;
 
+// add additional arguments into scenariowhitelist which will not be rendered
```
Comment (elalish) [inline packages/render-fidelity-tools/src/workflows/test-fidelity.ts:47]: These arguments are the whitelist, which means only these scenarios will be rendered. A blacklist would be a list of scenarios to not render.

File: packages/render-fidelity-tools/src/common.ts [hunk 7]
```diff
@@ -104,7 +104,7 @@ export interface ImageComparisonConfig {
 }
 
 export interface ComparableImage {
-  [index: number]: number;
+  [index: number]: number;  // what does this syntax mean?
```
Comment (elalish) [inline packages/render-fidelity-tools/src/common.ts:107]: This was fine as an excuse for a first PR, but is probably not the most efficient means of getting your questions answered about the code. I'd go ahead and revert this last commit and then we can merge this. Let's do a screen share and talk through code questions you have.
